src/rpc/net.cpp

```diff
@@ -322,6 +323,61 @@ static RPCHelpMan addnode()
     };
 }
 
+static RPCHelpMan addconnection()
+{
+    return RPCHelpMan{"addconnection",
+        "\nOpen an outbound connection to a specified node. This RPC is for testing only.\n",
+        {
+            {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The IP address and port to attempt connecting to."},
+            {"connection_type", RPCArg::Type::STR, RPCArg::Optional::NO, "Type of connection to open, either \"outbound\" or \"blockrelay\"."},
+        },
+        RPCResult{
+            RPCResult::Type::OBJ, "", "",
+            {
+                { RPCResult::Type::STR, "address", "Address of newly added connection." },
+                { RPCResult::Type::STR, "connection_type", "Type of connection opened." },
+            }},
+        RPCExamples{
+            HelpExampleCli("addconnection", "\"192.168.0.6:8333\" \"outbound\"")
+            + HelpExampleRpc("addconnection", "\"192.168.0.6:8333\" \"outbound\"")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+    if (Params().NetworkIDString() != CBaseChainParams::REGTEST) {
+        throw std::runtime_error("addconnection is for regression testing (-regtest mode) only.");
+    }
+
+    RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VSTR});
+    const std::string address = request.params[0].get_str();
+    const std::string conn_type_in{ToLower(TrimString(request.params[1].get_str()))};
```

Comment:
Many RPCs (sendtoaddress, sendmany, send, fundrawtransaction, walletcreatefundedpsbt, and bumpfee) have a case-insensitive string param, estimate_mode, that is also indicated as case insensitive in the various helps.

Reply:
With params like estimate_mode and connection_type (and addnode type too) where we ask users to supply a string word to select an option, I think it's reasonable to be lenient about capitalization as a thoughtful touch.

Reply:
I agree with @ajtowns. We should be strict about capitalization in rpc arguments. Adding a special case here doesn't make sense.

Reply:
I thnk `estimate_mode` only allows case-insensitivity because it initially required all-uppercase, see https://github.com/bitcoin/bitcoin/pull/11413#pullrequestreview-271355849 . Doesn't seem like a good example to replicate.

Reply:
I don't see the reason or UX advantage of case strictness for string words to select a choice, or even for the API endpoints/commands, but maybe that's from writing a number of clients that consumed APIs which were generally case indifferent (or from writing Lisp 😄).

Reply:
Ok, noted. Grepping a bit, it looks like `estimate_mode` is the only one.

Reply:
reverted to being case sensitive